### PR TITLE
Dev branch

### PR DESCRIPTION
**Pull request by jayahariv**

### CODE REVIEW

**Comment by jayahariv** [comment]
EE CI failure is due to the related EE PR.

File: Objective-C/Tests/DatabaseTest.m [hunk 1]
```diff
@@ -1346,13 +1347,14 @@ - (void) testDeletePurgedDocumentWithID {
     AssertNil([self.db documentWithID: documentID]);
     
     // Delete doc no-ops:
```
Comment (pasin) [inline Objective-C/Tests/DatabaseTest.m:1349]: This comment is not applicable anymore.

**Review by pasin — COMMENTED** [review]
Can you check if the build failure in PR validation is valid?

File: Objective-C/Tests/DatabaseTest.m (hunk 1 above)
Comment (jayahariv) [inline Objective-C/Tests/DatabaseTest.m:1349]:
```suggestion
    // Delete doc, 404 NotFound:
```

File: Objective-C/Tests/DatabaseTest.m [hunk 2]
```diff
@@ -880,8 +880,9 @@ - (void) testDeleteNonExistingDoc {
     }];
     
     // Delete doc1b, no-ops:
```
Comment (jayahariv) [inline Objective-C/Tests/DatabaseTest.m:882]:
```suggestion
    // Delete doc1b, 404 NotFound:
```

**Comment by jayahariv** [comment]
PR validation failure, is happening because the PR points to the master branch of EE, which doesn't get updated to the new socket changes. Which are in PRs only. 

```
[2021-06-23T15:34:20.098Z]         c4socket->nativeHandle = (__bridge_retained void*)self;
[2021-06-23T15:34:20.098Z]                 ^
//....
ios/vendor/couchbase-lite-core/C/include/c4Base.h:146:16: note: forward declaration of 'C4Socket'
[2021-06-23T15:34:20.098Z] typedef struct C4Socket C4Socket;
[2021-06-23T15:34:20.098Z]                ^
[2021-06-23T15:34:20.098Z] /Users/couchbase/workspace/eline_couchbase-lite-ios_PR-2856/Sources/Objective-C/Internal/Replicator/CBLMessageSocket.mm:221:58: error: member access into incomplete type 'C4Socket'
[2021-06-23T15:34:20.098Z]     auto cblSocket = (__bridge CBLMessageSocket*)c4Socket->nativeHandle;
[2021-06-23T15:34:20.098Z]                                                          ^
//.....
[2021-06-23T15:34:20.098Z] /Users/couchbase/workspace/eline_couchbase-lite-ios_PR-2856/Sources/Objective-C/Internal/Replicator/CBLMessageSocket.mm:307:25: error: member access into incomplete type 'C4Socket'
[2021-06-23T15:34:20.098Z]         CFRelease(socket->nativeHandle);
[2021-06-23T15:34:20.098Z]                         ^
```

**Review by pasin — APPROVED** [review]

**Pull request merged**